storage/redis/adapter.go

```diff
@@ -0,0 +1,101 @@
+package redis
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	r "gopkg.in/redis.v3"
+)
+
+type adapter struct {
+	bufferSize  int
+	redisClient *r.Client
+}
+
+// NewStorageAdapter returns a pointer to a new instance of a redis-based storage.Adapter.
+func NewStorageAdapter(bufferSize int) (*adapter, error) {
```

Comment:
So... I tried to make it return a `storage.Adapater` and was prompty reminded of why I don't. It would cause an import cycle and wouldn't compile. The the `storage` package references the `redis` package that this implementation resides in, so the `redis` package must not import `storage`.

I could probably fix that by putting all the implementations in a single package along with the interface they all implement. (Being more experiences with Go than I was when I created that interface last fall, this seems to be a more favorable approach for the simple fact that it allows for what you're requesting without causing a import cycle.)

All that said, I would _love_ to do that, but would consider it a more general refactor as it would touch all three implementations. If I commit to making that change in a separate PR, can we unblock this?